### PR TITLE
[ONNX] Add ONNX Compress operator.

### PR DESCRIPTION
**Pull request by tsocha**
### Details:
 - Add support for Compress operator in ONNX Importer
 - Add tests for this Operator which uses Constant Input
 - Fix NonZero validation and shape inference.

### Tickets:
 - 33644

### CODE REVIEW

File: ngraph/frontend/onnx_import/src/op/non_zero.cpp [hunk 1]
```diff
@@ -17,7 +17,11 @@ namespace ngraph
             {
                 OutputVector non_zero(const Node& node)
                 {
-                    const auto data = node.get_ng_inputs().at(0);
+                    auto data = node.get_ng_inputs().at(0);
+                    if (data.get_element_type() == element::boolean)
+                    {
+                        data = std::make_shared<default_opset::Convert>(data, element::u8);
```
Comment (tomdol) [inline ngraph/frontend/onnx_import/src/op/non_zero.cpp:23]: Is there an onnx importer test for boolean inputs that validates it?

File: ngraph/test/onnx/onnx_import_with_editor.in.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#include "ngraph/ngraph.hpp"
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
+#include "util/engine/test_engines.hpp"
+
+using namespace ngraph;
+
+static std::string s_manifest = "${MANIFEST}";
+
+using Inputs = std::vector<std::vector<float>>;
```
Comment (tomdol) [inline ngraph/test/onnx/onnx_import_with_editor.in.cpp:25]: I believe this is a copy-and-paste artifact but we don't use those aliases any more. Not in this file at least.

**Review by tomdol — CHANGES_REQUESTED** [review]

File: ngraph/frontend/onnx_import/src/op/non_zero.cpp (hunk 1 above)
Comment (tsocha) [reply ngraph/frontend/onnx_import/src/op/non_zero.cpp:23]: I have no information about such case but bool type is supported in the ONNX spec while OV NonZero accept only numeric types.

File: ngraph/test/onnx/onnx_import_with_editor.in.cpp (hunk 2 above)
Comment (tsocha) [reply ngraph/test/onnx/onnx_import_with_editor.in.cpp:25]: `s_manifest` is needed for test skipping from manifest file AFAIK

Comment (tomdol) [reply ngraph/test/onnx/onnx_import_with_editor.in.cpp]: I was referring to Inputs and Outputs aliases, no the string above.

**Review by tomdol — APPROVED** [review]

**Review by postrational — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**